src/device-state/target-state.ts

```diff
@@ -127,6 +127,13 @@ export const update = async (
 			);
 		}
 
+		// If this update request came from the API reset emitted to cancel
+		// any processes that may be already applying the cached target release
+		// such as the back off timeout for when update locks are detected
+		if (isFromApi) {
```

Comment:
There is code that already exists which will cancel and immediately try to apply the target state again. It was just never used to my knowledge and this change uses that. I think this code was intended to always exist tbh since the apply target state cancellation seems to never be called.

Reply:
This doesn't make the `statusCode === 304` as it still compares the local target state and the one the server has.